### PR TITLE
Add a progress reporter to the native indexing method

### PR DESCRIPTION
**Pull request by larslj**
A small patch to add a spinning progress reporter to the native indexing method.

### CODE REVIEW

**Comment by bbatsov** [comment]
Some visual demonstration would be helpful (e.g. animated gif).

**Comment by larslj** [comment]
The progress reporter (introduced in Emacs 24.1) will show a spinning wheel in the echo area, in this case right after the text "Projectile is indexing %s". The spinning wheel is cycling through the characters "-\|/".

File: projectile.el [hunk 1]
```diff
@@ -714,11 +714,12 @@ Files are returned as relative paths to the project root."
...
 (defun projectile-dir-files-native (root directory)
   "Get the files for ROOT under DIRECTORY using just Emacs Lisp."
-  (message "Projectile is indexing %s. This may take a while."
-           (propertize directory 'face 'font-lock-keyword-face))
-  ;; we need the files with paths relative to the project root
-  (-map (lambda (file) (file-relative-name file root))
-        (projectile-index-directory directory (projectile-patterns-to-ignore))))
+  (let ((progress (make-progress-reporter
+	 (format "Projectile is indexing %s" (propertize directory
+					   'face 'font-lock-keyword-face)))))
```
Comment (bbatsov) [inline projectile.el]: The indentation here is incorrect.

File: projectile.el [hunk 2]
```diff
@@ -714,11 +714,12 @@ Files are returned as relative paths to the project root."
...
+    ;; we need the files with paths relative to the project root
+    (-map (lambda (file) (file-relative-name file root))
+	  (projectile-index-directory directory (projectile-patterns-to-ignore) progress))))
```
Comment (bbatsov) [inline projectile.el]: Here as well.

File: projectile.el [hunk 3]
```diff
@@ -714,11 +714,12 @@ Files are returned as relative paths to the project root."
 
 (defun projectile-dir-files-native (root directory)
   "Get the files for ROOT under DIRECTORY using just Emacs Lisp."
-  (message "Projectile is indexing %s. This may take a while."
-           (propertize directory 'face 'font-lock-keyword-face))
-  ;; we need the files with paths relative to the project root
-  (-map (lambda (file) (file-relative-name file root))
-        (projectile-index-directory directory (projectile-patterns-to-ignore))))
+  (let ((progress (make-progress-reporter
```
Comment (bbatsov) [inline projectile.el]: I'd suggest renaming progress to progress-reporter.

File: CHANGELOG.md [hunk 4]
```diff
@@ -34,6 +34,7 @@ buffers by name
 * Collect search tool (`grep`, `ag`, `ack`) keybindings under a common keymap prefix (`C-c p s`)
 * Remove `defcustom` `projectile-remember-window-configs` in favor of
   `persp-projectile.el`.
+* Progress reporter for the native indexing method
```
Comment (bbatsov) [inline CHANGELOG.md]: This sentence is missing its `.`.

**Comment by bbatsov** [comment]
Squash the two commits into one and we'll be good to go.

**Comment by larslj** [comment]
Done. Thanks for your patience. I'm not that used to git and github.

**Comment by bbatsov** [comment]
:+1: Thanks for your contribution! :-)

**Pull request merged**